fecgov/fec-cms
# 4413 - Debts datatable

## Summary

- Resolves #4413 

Adding the debts datatable and links, behind a feature flag.

### Required reviewers

- UX
- Front-end
- Back-end?
- SME (not on localhost)
- Content? (not on localhost)

## Impacted areas of the application

- /data/browse-data/
- /data/debts/
- committee profile pages (or anywhere a link to the debts tables will be)

## Screenshots

The browse-data page (Debts and "and debts" will only appear when the debts feature flag is on)
![image](https://github.com/fecgov/fec-cms/assets/26720877/6e36b239-5181-4091-8576-c5e3a93d6072)

The debts tables (only available when the debts feature flag is on)
![image](https://github.com/fecgov/fec-cms/assets/26720877/8d71cde3-178b-4d04-954e-a2e15a9b4cf8)

The bottom of a committee profile page. The debts-owed-by should only be a link if the feature flag is on—the numbers haven't been affected
![image](https://github.com/fecgov/fec-cms/assets/26720877/d9abfb4c-a407-4161-8d25-c538350b62b0)


## Related PRs

Related PRs against other branches:

branch | PR
------ | ------
fix/other_pr | [link]()
feature/other_pr | [link]()

## KNOWN ISSUES / QUESTIONS
- I need content for [the Debts section](http://127.0.0.1:8000/data/browse-data/?tab=loans-debts)
- Some candidate profiles don't have `committee_id` when building their links so I've made that link conditional in that `?committee_id=` only gets added if `committeed_id` exists


## How to test

- Pull the branch
- `npm i`
- `npm run build`
- `./manage.py runserver`
- Check the [Browse data](http://127.0.0.1:8000/data/browse-data/) page
- Check the [Loans and debts](http://127.0.0.1:8000/data/browse-data/?tab=loans-debts) tab/section of that page
- Click over to the [Debts datatable](http://127.0.0.1:8000/data/debts/)
- Play with the filters
- Click the data rows or their icon in the right column
- Click to a committee profile
- Check the profile's Financial summary section (,review the content generally?), check the debts/loans section at the bottom (value calculation hasn't changed, but the owed-by numbers should be a link now
- Click the owed-by link to make sure the datatable is filtering properly
- Extra credit: if you'd like to test the feature flag, edit `fec/fec/settings/base.py` and
   - edit line 90 (`FEATURES['debts']`) to be `False`, or
   - comment-out that line and line 72 (`'debts': `…)
